recipes/libsigcpp/2.x.x/conanfile.py

```diff
@@ -0,0 +1,113 @@
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires("meson/0.59.1")
+        self.build_requires("pkgconf/1.7.4")
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version],
+            strip_root=True,
+            destination=self._source_subfolder
+        )
+
+    def build(self):
+        if not self.options.shared:
+            tools.replace_in_file(
+                os.path.join(self._source_subfolder, "sigc++config.h.meson"),
+                "define SIGC_DLL 1", "undef SIGC_DLL")
+        with tools.environment_append(tools.RunEnvironment(self).vars):
+            meson = self._configure_meson()
+            meson.build()
+
+    def _configure_meson(self):
+        meson = Meson(self)
+        defs = {}
+        defs["build-examples"] = "false"
+        defs["build-documentation"] = "false"
+        defs["default_library"] = "shared" if self.options.shared else "static"
+        meson.configure(
+            defs=defs,
+            build_folder=self._build_subfolder,
+            source_folder=self._source_subfolder,
+            pkg_config_paths=[self.install_folder],
+        )
+        return meson
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        meson = self._configure_meson()
+        meson.install()
+        if self.settings.compiler == "Visual Studio":
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.pdb")
+            if not self.options.shared:
+                rename(self,
+                       os.path.join(self.package_folder, "lib", "libsigc-2.0.a"),
+                       os.path.join(self.package_folder, "lib", "sigc-2.0.lib"))
+
+        for header_file in glob.glob(os.path.join(self.package_folder, "lib", "sigc++-2.0", "include", "*.h")):
+            shutil.move(
+                header_file,
+                os.path.join(self.package_folder, "include",
+                             "sigc++-2.0", os.path.basename(header_file))
+            )
+
+        for dir_to_remove in ["pkgconfig", "sigc++-2.0"]:
+            tools.rmdir(os.path.join(
+                self.package_folder, "lib", dir_to_remove))
+
+    def package_info(self):
+        self.cpp_info.components["sigc++-2.0"].names["pkg_config"] = "sigc++-2.0"
+        self.cpp_info.components["sigc++-2.0"].includedirs.append(os.path.join("include", "sigc++-2.0"))
+        self.cpp_info.components["sigc++-2.0"].libs = tools.collect_libs(self)
```

Comment:
Try to avoid collect_libs, it's not transparent and users does not know the exported library name. Also, it uses alphabetic order to link libraries, which may break static link.